### PR TITLE
[21745] Add QoS getters from raw XML

### PR DESCRIPTION
**Pull request by juanlofer-eprosima**
## Description

Update documentation covering new API: https://github.com/eProsima/Fast-DDS/pull/5256

## Contributor Checklist

- [x] Commit messages follow the project guidelines. 
- [x] Code snippets related to the added documentation have been provided.
- [x] Documentation tests pass locally.
- _N/A_ Applicable backports have been included in the description.

## Reviewer Checklist

- [ ] The PR has a milestone assigned.
- [ ] The title and description correctly express the PR's purpose.
- [ ] Check contributor checklist is correct.
- [ ] CI passes without warnings or errors.

### CODE REVIEW

File: docs/fastdds/xml_configuration/making_xml_profiles.rst [hunk 1]
```diff
@@ -78,6 +78,100 @@ Please, refer to :ref:`xml_profiles` for further information regarding loading p
...
+
+Get QoS from raw XML profiles
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Although the :ref:`standard procedure <loadingapplyingprofiles>` is to first load XML profiles, and then create entities
+given the profile name of choice, it is also possible to get the desired QoS from the XML profile directly so it can be
+modified before being used.
+For this purpose, the following public methods are available:
+
+.. collapse:: DomainParticipantFactory
```
Comment (juanlofer-eprosima) [inline docs/fastdds/xml_configuration/making_xml_profiles.rst:91]: Don't know how to present this information in a clear and fitting manner. These are two alternative versions none of which I really like.

File: docs/fastdds/xml_configuration/making_xml_profiles.rst [hunk 2]
```diff
@@ -78,6 +78,100 @@ Please, refer to :ref:`xml_profiles` for further information regarding loading p
...
+----------
+
+.. collapse:: Subscriber
+
+    - |Subscriber::get_datareader_qos_from_xml-api|
+    - |Subscriber::get_default_datareader_qos_from_xml-api|
+
+----------
+
+- |DomainParticipantFactory-api|
```
Comment (rsanchez15) [inline docs/fastdds/xml_configuration/making_xml_profiles.rst:129]: I prefer this version.

File: docs/fastdds/xml_configuration/making_xml_profiles.rst [hunk 3]
```diff
@@ -78,6 +78,100 @@ Please, refer to :ref:`xml_profiles` for further information regarding loading p
...
+
+    - |DomainParticipantFactory::get_participant_qos_from_xml-api|
```
Comment (rsanchez15) [inline docs/fastdds/xml_configuration/making_xml_profiles.rst:131]: All these links are not working as the API reference seems to not be updated.

**Review by rsanchez15 — CHANGES_REQUESTED** [review]

File: docs/fastdds/xml_configuration/making_xml_profiles.rst (hunk 3 above)
Comment (juanlofer-eprosima) [reply docs/fastdds/xml_configuration/making_xml_profiles.rst:131]: It is working locally...

Comment (MiguelCompany) [reply docs/fastdds/xml_configuration/making_xml_profiles.rst:93]: The RTD build would probably use Fast DDS master branch, so the API reference would not be there till we merge the implementation PR

**Review by rsanchez15 — APPROVED** [review]

**Pull request merged**